### DISCUSSION TITLE
语句流输入方式还是不够方便，并不像整句输入

### DISCUSSION THREAD

**Discussion by git-zeta** [rime/home#1028] [General]
我安装Rime完全是为了获得类似win7以前的旧版微软拼音的整句输入效果，但是用下来发现，语句流输入方式并没有真正符合我的需求。

真正的整句输入，应该是在一次性不选字连续输完整个句子之后，可以移动光标到句子中间的任意一个字词处进行重选或修改或插入，而不会扰乱其前后的其余字词，这样往往只需要重选或修改或插入一两个字词就能完成整个句子的输入，是整句输入模式最大的优势。而自从微软拼音长句模式被微软砍掉之后，至今我找到的所有号称支持整句输入的输入法都没有实现上述关键功能。Rime也不例外，输完句子之后如果把光标移到句子中间，就会导致后半句的全部字词都必须逐一重选，反而比分词输入的方式更费时间了。这其实还是把句子当成了超长词组来处理，根本没有专门考虑整句输入的基本逻辑需求。

我不知道微软拼音的长句模式到底有多难实现，明明是那么方便好用的功能为什么没有人继承下来？真心希望Rime能重现这种输入方式。

**Comment by lotem** [comment]
有些夸张了。实际的情况是比起词组和短句的输入方式并不算更方便。大部分用家的选择已经说明问题，开发者也是依据事实、根据用户的需要设计产品。

Rime为支持整句输入方式做了必要设计，但作者可投入在这个项目上的时间有限，只能先实现更有用的部分，并没有实现一套完整的整句输入算法。情怀部分和少数派偏好，就留给大家来扩展。这正是本项目开源的目的。

**Reply by git-zeta** [reply]
我只求一个功能，就是左右移动光标时，不要导致撤消已经自动或手动选好的字词。请问这个功能能否通过自行修改设置实现？

比如：框中显示“明显彼此组输入效率高得多”时按两下ctrl+右，能不能不要变成“mingxian^bicizushuruxiaolvgaodeduo”，而是变成“明显^bi此组输入效率高得多”

再比如：框中显示“右方^xiangjian”时按一下左，能不能不要变成“you^fangxiangjian”，而是变成“右^fangxiangjian”

**Reply by ChihSee-Hsie** [reply]
日语输入法是不是就这样做的？整句输入之后按下空格，在转换的同时自动切成几个词组，移动光标可以改动个别词组那种。

**Reply by ShikiSuen** [reply]
只是 Unigram 的话，窃以为「真正的整句输入」是个伪命题。
Bigram 我就不清楚了。

P.S.: 旧版微软拼音的行为及背后的逻辑，恐怕只能问当年参与研发的团队。
